File: lab4/Java/TechLead.java
public class TechLead extends Funcionario implements InterfaceDesenvolvedor, InterfaceGerente {
    private String linguagemPrincipal;
    private int qtdEquipes;
    private int qtdProjetos;

    public TechLead(String nomeFuncionario, float salarioFuncionario, int qtdEquipes,  String linguagemPrincipal, int qtdProjetos) {
        super(nomeFuncionario, salarioFuncionario);
        this.linguagemPrincipal = linguagemPrincipal;
        this.qtdEquipes = qtdEquipes;
        this.qtdProjetos = qtdProjetos;
    }

    public String getLinguagemPrincipal() {
        return linguagemPrincipal;
    }

    public String projetosRealizados() {
        return "TechLead realizou " +qtdProjetos + " projetos na linguagem: " + linguagemPrincipal;
    }
    
    public int getQtdEquipe() {
        return qtdEquipes;
    }

    public float bonus() {
        return qtdEquipes * 500;
    }

    @Override
    public String descricao() {
        return super.descricao() + ", Linguagem Principal: " + linguagemPrincipal + ", Equipes: " + qtdEquipes + ", Bônus: " + bonus();
    }   
    
}

File: lab4/C++/TechLead.cpp
#include "TechLead.hpp"

TechLead::TechLead(string nomeFuncionario, float salarioBase, int numero_de_equipes, string linguagemPrincipal, int projetos_em_andamento)
    : Funcionario(nomeFuncionario, salarioBase),
    Gerente(nomeFuncionario, salarioBase, numero_de_equipes),
    Desenvolvedor(nomeFuncionario, salarioBase, linguagemPrincipal),
    projetos_em_andamento(projetos_em_andamento)
{}
string TechLead::descricao(){
    return "TechLead: Nome: " + nomeFuncionario + 
           ", Salário base: " + to_string(salarioBase) + 
           ", Número de equipes: " + to_string(numero_de_equipes) + 
           ", Linguagem principal: " + linguagemPrincipal +
           ", Projetos em andamento: " + to_string(projetos_em_andamento) + ".";
}